### PR TITLE
Fix clang++ compiler warnings

### PR DESCRIPTION
**Pull request by peterbarker**
./Tools/autotest/sim_vehicle.py --waf-configure-arg="--check-cxx-compiler=clang++ --check-c-compiler=clang" -v ArduCopter --gdb --debug  -v ArduPlane

This incorporates a fix to compiler flags for clang++.

Most contentious thing here is doing as the compiler suggests and using `std::abs(...)`

### CODE REVIEW

File: libraries/AP_OpticalFlow/AP_OpticalFlow_Pixart.cpp [hunk 1]
```diff
@@ -309,11 +309,14 @@ void AP_OpticalFlow_Pixart::timer(void)
     }
     
 #if 0
+    static uint32_t last_print_ms;
```
Comment (amilcarlucas) [inline libraries/AP_OpticalFlow/AP_OpticalFlow_Pixart.cpp]: Shouldn't statics be avoided at all cost ?

File: libraries/AP_Math/AP_Math.cpp [hunk 2]
```diff
@@ -28,7 +28,7 @@ is_equal(const Arithmetic1 v_1, const Arithmetic2 v_2)
     if (std::is_same<double, common_type_nonconst>::value) {
         return fabs(v_1 - v_2) < std::numeric_limits<double>::epsilon();
     }
-    return fabsf(v_1 - v_2) < std::numeric_limits<float>::epsilon();
+    return std::abs(v_1 - v_2) < std::numeric_limits<float>::epsilon();
```
Comment (khancyr) [inline libraries/AP_Math/AP_Math.cpp]: std::abs should work the same that fabsf, I don't understand why clang trigger a warning. I case of double it should reach this, no ?

**Comment by khancyr** [comment]
LGTM, but it break on fmuv1

**Comment by peterbarker** [comment]
On Wed, 28 Feb 2018, Amilcar Lucas wrote:

>  #if 0
> +    static uint32_t last_print_ms;
> 
> Shouldn't statics be avoided at all cost ?

Within #if 0'd debug statements is reasonable.

**Comment by peterbarker** [comment]
On Wed, 28 Feb 2018, Pierre Kancir wrote:

> LGTM, but it break on fmuv1

Geeze.

I've removed the std::abs stuff from this PR.

**Comment by khancyr** [comment]
ok ! it is good to go now !

**Comment by rmackay9** [comment]
merged!

**Pull request merged**